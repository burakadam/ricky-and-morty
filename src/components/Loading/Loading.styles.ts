import { keyframes } from '@emotion/react';
import styled from '@emotion/styled';

const fadeInAndOut = keyframes`
  0% {
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
  100%{opacity:0}
`;

const Container = styled('div')`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 100%;
  height: 50vh;
  animation: ${fadeInAndOut} 1.5s infinite;
`;

const Logo = styled('img')`
  height: 40px;
  width: auto;
  border-radius: 50%;
  margin-bottom: 20px;
`;

export { Container, Logo };
