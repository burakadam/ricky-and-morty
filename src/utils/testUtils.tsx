import type { PreloadedState } from '@reduxjs/toolkit';
import type { RenderOptions } from '@testing-library/react';
import { render } from '@testing-library/react';
import type { PropsWithChildren } from 'react';
import React from 'react';
import { Provider } from 'react-redux';

import type { AppStore, RootState } from '@/store/store';
import { setupStore } from '@/store/store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  // eslint-disable-next-line @typescript-eslint/no-shadow
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export function Wrapper({ children }: { children: React.ReactElement }) {
  return <>{children}</>;
}

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';
export { customRender as render };
