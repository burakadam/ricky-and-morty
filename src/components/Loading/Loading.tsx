import { Typography } from '@mui/material';

import LOGO from '@/assets/images/rick-and-morty-silhouette.jpg';

import * as S from './Loading.styles';

const Loading = () => (
  <S.Container>
    <S.Logo src={LOGO} alt="Rick and Morty silhoutte" />
    <Typography>LOADING</Typography>
  </S.Container>
);

export { Loading };
