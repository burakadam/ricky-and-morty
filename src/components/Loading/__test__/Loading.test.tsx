import { render } from '@/utils/testUtils';

import { Loading } from '../Loading';

describe('loading component', () => {
  let container: HTMLElement;

  beforeEach(() => {
    // eslint-disable-next-line testing-library/no-render-in-setup
    ({ container } = render(<Loading />));
  });

  it('render component correctly', () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  it('has loading animation', () => {
    const loader = container.querySelector('div') as HTMLElement;
    const loaderStyle = window.getComputedStyle(loader).animation;

    expect(loaderStyle).toEqual('animation-116j0z6 1.5s infinite');
  });
});
