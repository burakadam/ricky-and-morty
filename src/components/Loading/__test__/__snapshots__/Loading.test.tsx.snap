// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading component render component correctly 1`] = `
<div
  class="css-smbw90"
>
  <img
    alt="Rick and Morty silhoutte"
    class="css-1cg9ond"
    src="rick-and-morty-silhouette.jpg"
  />
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    LOADING
  </p>
</div>
`;
